From 818dbad4f252b284fe5f974e71850a8636db7c85 Mon Sep 17 00:00:00 2001
From: Liqiang Tao <taolq@outlook.com>
Date: Fri, 9 Apr 2021 10:46:48 +0800
Subject: [PATCH] [riscv64] Block trampoline pool in Ror and Dror macro
 assembler

Change-Id: Ia88e43711d54e1aa651757f6a2bac7005b4274aa
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/2814129
Reviewed-by: Brice Dobry <brice.dobry@futurewei.com>
Commit-Queue: Leszek Swirski <leszeks@chromium.org>
Cr-Commit-Position: refs/heads/master@{#73897}
---
 src/codegen/riscv64/macro-assembler-riscv64.cc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/codegen/riscv64/macro-assembler-riscv64.cc b/src/codegen/riscv64/macro-assembler-riscv64.cc
index 5d1c065981..df0903d31d 100644
--- a/src/codegen/riscv64/macro-assembler-riscv64.cc
+++ b/src/codegen/riscv64/macro-assembler-riscv64.cc
@@ -884,6 +884,7 @@ void TurboAssembler::Sll64(Register rd, Register rs, const Operand& rt) {
 void TurboAssembler::Ror(Register rd, Register rs, const Operand& rt) {
   UseScratchRegisterScope temps(this);
   Register scratch = temps.Acquire();
+  BlockTrampolinePoolScope block_trampoline_pool(this);
   if (rt.is_reg()) {
     negw(scratch, rt.rm());
     sllw(scratch, rs, scratch);
@@ -908,6 +909,7 @@ void TurboAssembler::Ror(Register rd, Register rs, const Operand& rt) {
 void TurboAssembler::Dror(Register rd, Register rs, const Operand& rt) {
   UseScratchRegisterScope temps(this);
   Register scratch = temps.Acquire();
+  BlockTrampolinePoolScope block_trampoline_pool(this);
   if (rt.is_reg()) {
     negw(scratch, rt.rm());
     sll(scratch, rs, scratch);
-- 
2.35.1

